docs/fast_build.md

```diff
@@ -1,4 +1,113 @@
-Optimizing a Tiltfile
+Expanding a Tiltfile
 =====================
 
-TODO(nick): Write this guide.
+This tutorial looks at a `Tiltfile` with build optimizations.
+We explain what they do, and why you would want to use them.
+
+In the previous tutorial, we introduced the `static_build()` function.
+This function builds a Docker image. Tilt will watch the inputs to the
+image, and rebuild it every time they change.
+
+This works well for interpreted languages like JavaScript and Python
+where you can add the files and go. For servers that need to be compiled,
+it would be too slow to recompile from scratch every time.
+
+That's why Tilt has a function `start_fast_build()` for lightning-fast local
+Kubernetes development.
+
+Let's look at an example in the [tiltdemo repo](https://github.com/windmilleng/tiltdemo):
+
+```
+git clone https://github.com/windmilleng/tiltdemo
+cd tiltdemo
+```
+
+The `Tiltfile` at the root of the repo contains this function:
+
+```
+# -*- mode: Python -*-
+
+def tiltdemo1():
+  yaml = read_file('deployments/demoserver1.yaml')
+  image_name = 'gcr.io/windmill-test-containers/tiltdemo/demoserver1'
+  start_fast_build('Dockerfile', image_name, '/go/bin/demoserver1')
+  repo = local_git_repo('.')
+  add(repo.path('cmd/demoserver1'),
+      '/go/src/github.com/windmilleng/tiltdemo/cmd/demoserver1')
+  run('go install github.com/windmilleng/tiltdemo/cmd/demoserver1')
+  image = stop_build()
+  return k8s_service(image, yaml)
+```
+
+This looks similar to the `Tiltfile` in previous tutorials, but instead of building
+with `static_build()`, it contains `start_fast_build()` and `stop_build()`. Let's zoom
+in on that part of the function.
+
+
+```
+start_fast_build('Dockerfile', image_name, '/go/bin/demoserver1')
+repo = local_git_repo('.')
+add(repo.path('cmd/demoserver1'),
+    '/go/src/github.com/windmilleng/tiltdemo/cmd/demoserver1')
+run('go install github.com/windmilleng/tiltdemo/cmd/demoserver1')
+image = stop_build()
+```
+
+These lines configure `tilt` to do incremental image builds. We'll step through it line-by-line.
+
+* `start_fast_build('Dockerfile', image_name, '/go/bin/demoserver1')`
+
+`start_fast_build` begins the build.
+This is setting up the build environment before we add any code.
+We build on top of the image in `Dockerfile`. Our new
```

Comment:
Should we explain here what things cannot be in this "base" Dockerfile?

Reply:
good idea! added a line about this